api/src/wfl/module/covid.clj

```diff
@@ -207,17 +209,55 @@
         (jdbc/execute! tx [(format query queue) id])))
     (throw (ex-info "No snapshots in queue" {:source source}))))
 
+(defn ^:private go-create-snapshot!
+  "Create snapshot in TDR from `dataset` body, `table`
+   `now` string and `row-ids` then write job info as well
+   as rows into `source-details-name` table."
+  [now dataset table row-ids]
+  (let [columns     (-> (datarepo/all-columns dataset table)
+                        (->> (map :name) set)
+                        (conj "datarepo_row_id"))
+        job-id (-> (datarepo/make-snapshot-request dataset columns table row-ids)
+                   (assoc :name now)
+                   (datarepo/create-snapshot-job))]
+    job-id))
+
 ;; Create and add new snapshots to the snapshot queue
-(defn ^:private update-tdr-source [source]
-  (letfn [(find-new-rows       [source now] [])
-          (make-snapshots      [source row-ids])
-          (write-snapshots     [source snapshots])
-          (update-last-checked [now])]
-    (let [now     (OffsetDateTime/now)
-          row-ids (find-new-rows source now)]
-      (when-not (empty? row-ids)
-        (write-snapshots source (make-snapshots source row-ids)))
-      (update-last-checked source now))))
+(defn ^:private update-tdr-source
+  "Check for new data in TDR, create new snapshots, insert
+   resulting job creation ids into database and update the
+   timestamp for next time using transaction TX."
+  [{:keys [id dataset dataset_table table_column_name last_checked details] :as _source}]
+  (letfn [(find-new-rows [now]
+            (-> (datarepo/query-table-between
+                 dataset
+                 dataset_table
+                 table_column_name
+                 [last_checked now]
+                 [:datarepo_row_id])
+                :rows
+                flatten))
+          (make-snapshots [now-obj row-ids]
+            (let [shards (partition-all 500 row-ids)
+                  compact-now (.format now-obj (DateTimeFormatter/ofPattern "YYYYMMDD'T'HHMMSS"))]
+              (doall (map #(go-create-snapshot! compact-now dataset dataset_table %) shards))))
+          (write-snapshots-creation-jobs [tx now row-ids snapshots-creation-jobs]
+            (jdbc/insert-multi! tx
+                                details
+                                (map (fn [id] {:snapshot_creation_job_id id
```

Comment:
I'm sorry for missing this in your last PR, but I believe that `snapshot_id` was inadvertently removed from the source details schema, it is still in the LucidChart:

https://lucid.app/lucidchart/63bc90dd-5db6-4611-adef-be87d0e8b6e3/edit?page=7Wftl_-jXJbR#

I can add it back in a small PR but have a few questions:
- Will you be responsible for updating the `snapshot_id` in the source details record? 
- In checking the source queue for available snapshots, I need to find the first unconsumed record with an available snapshot.  Let's talk about this when you're back so that we don't cover the same ground.

Reply:
AH nice catch, I completely missed that field somehow… I will add it back in a later commit!

re your questions, I think we should defo chat cauz I'm not sure who should be checking and updating the source queue for available snapshots now.

Reply:
Fixed